src/Tracing/SentryWebTransactionAdapter.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry\Tracing;
+
+use Psr\Log\LoggerInterface;
+use Sentry\SentrySdk;
+use Sentry\Tracing\SpanContext;
+use Sentry\Tracing\TransactionContext;
+use Yiisoft\Log\Logger;
+
+class SentryWebTransactionAdapter
+{
+    protected ?Logger $logger = null;
+
+    public function __construct(LoggerInterface $logger, private SentryTraceMiddleware $middleware)
+    {
+        if ($logger instanceof Logger) {
```

Comment:
it looks sad that you are trying to inject object by interface but using only specific children.